From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Fri, 15 Mar 2019 22:54:59 +0100
Subject: [PATCH] {MAYBE UPS} drivers/net/ethernet/mellanox: silence
 -Wint-in-bool-context
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

gcc 8.2.1 reports:

    In file included from drivers/net/ethernet/mellanox/mlxsw/core.h:17,
                     from drivers/net/ethernet/mellanox/mlxsw/core_env.c:7:
    drivers/net/ethernet/mellanox/mlxsw/core_env.c: In function
    ‘mlxsw_env_module_temp_thresholds_get’:
    drivers/net/ethernet/mellanox/mlxsw/reg.h:8015:45: error: ‘*’ in
    boolean context, suggest ‘&&’ instead [-Werror=int-in-bool-context]
     #define MLXSW_REG_MTMP_TEMP_TO_MC(val) (val * 125)
                                            ~~~~~^~~~~~
    drivers/net/ethernet/mellanox/mlxsw/core_env.c:116:8: note: in
    expansion of macro ‘MLXSW_REG_MTMP_TEMP_TO_MC’
       if (!MLXSW_REG_MTMP_TEMP_TO_MC(module_temp)) {
            ^~~~~~~~~~~~~~~~~~~~~~~~~
---
 drivers/net/ethernet/mellanox/mlxsw/core_env.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/mellanox/mlxsw/core_env.c b/drivers/net/ethernet/mellanox/mlxsw/core_env.c
index 7a15e932ed2f..c1c1965d7acc 100644
--- a/drivers/net/ethernet/mellanox/mlxsw/core_env.c
+++ b/drivers/net/ethernet/mellanox/mlxsw/core_env.c
@@ -113,7 +113,7 @@ int mlxsw_env_module_temp_thresholds_get(struct mlxsw_core *core, int module,
 		return 0;
 	default:
 		/* Do not consider thresholds for zero temperature. */
-		if (!MLXSW_REG_MTMP_TEMP_TO_MC(module_temp)) {
+		if (MLXSW_REG_MTMP_TEMP_TO_MC(module_temp) == 0) {
 			*temp = 0;
 			return 0;
 		}
-- 
